env/HERA.env

```diff
@@ -188,43 +188,11 @@ elif [[ "${step}" = "eupd" ]]; then
     export USE_CFP=${USE_CFP:-"YES"}
     export APRUNCFP="${launcher} -n \$ncmd ${mpmd_opt}"
 
-elif [[ "${step}" = "fcst" ]]; then
+elif [[ "${step}" = "fcst" ]] || [[ "${step}" = "efcs" ]]; then
 
-    #PEs and PEs/node can differ for GFS and GDAS forecasts if threading differs
-    if [[ ${CDUMP} == "gfs" ]]; then
-        npe_fcst=${npe_fcst_gfs}
-        npe_node_fcst=${npe_node_fcst_gfs}
-        nth_fv3=${nth_fv3_gfs}
-    fi
-
-    nth_max=$((npe_node_max / npe_node_fcst))
-
-    export NTHREADS_FV3=${nth_fv3:-${nth_max}}
-    [[ ${NTHREADS_FV3} -gt ${nth_max} ]] && export NTHREADS_FV3=${nth_max}
-    export cores_per_node=${npe_node_max}
-    export APRUN_FV3="${launcher} -n ${npe_fcst}"
-
-    export NTHREADS_REGRID_NEMSIO=${nth_regrid_nemsio:-1}
-    [[ ${NTHREADS_REGRID_NEMSIO} -gt ${nth_max} ]] && export NTHREADS_REGRID_NEMSIO=${nth_max}
-    export APRUN_REGRID_NEMSIO="${launcher} -n ${LEVS}"
-
-    export NTHREADS_REMAP=${nth_remap:-2}
-    [[ ${NTHREADS_REMAP} -gt ${nth_max} ]] && export NTHREADS_REMAP=${nth_max}
-    export APRUN_REMAP="${launcher} -n ${npe_remap:-${npe_fcst}}"
-    export I_MPI_DAPL_UD="enable"
-
-elif [[ "${step}" = "efcs" ]]; then
-
-    nth_max=$((npe_node_max / npe_node_efcs))
-
-    export NTHREADS_FV3=${nth_efcs:-${nth_max}}
-    [[ ${NTHREADS_FV3} -gt ${nth_max} ]] && export NTHREADS_FV3=${nth_max}
-    export cores_per_node=${npe_node_max}
-    export APRUN_FV3="${launcher} -n ${npe_efcs}"
-
-    export NTHREADS_REGRID_NEMSIO=${nth_regrid_nemsio:-1}
-    [[ ${NTHREADS_REGRID_NEMSIO} -gt ${nth_max} ]] && export NTHREADS_REGRID_NEMSIO=${nth_max}
-    export APRUN_REGRID_NEMSIO="${launcher} -n ${LEVS}"
+    [[ "${CDUMP}" =~ "gfs" ]] && nprocs="npe_${step}_gfs" || nprocs="npe_${step}"
```

Comment:
Can you use a standard if/else here? IMO short-circuits are already harder to parse, and here you also have to know the operator precedence.

Reply:
Any reason to do it this way instead of making `nprocs` a local?

Reply:
done on expansion.
And because this is not a function. `local` has no meaning outside a function (pretty sure about that, but not 100%)